src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -579,14 +579,26 @@ private void ReadToken(string encodedJson)
             EncodedToken = encodedJson;
         }
 
+        /// <summary>
+        /// Serializes the <see cref="JsonWebToken"/>.
+        /// </summary>
+        /// <returns>A string containing the header and payload in JSON format.</returns>
+        public override string ToString()
+        {
+            if (!IsEncrypted)
+                return Header.ToString() + "." + Payload.ToString();
```

Comment:
In most cases EncodedHeader and EncodedPayload is what you want here.
If you fix this bug: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2065 then you will always get proper value.